documentation/src/main/asciidoc/userguide/chapters/domain/associations.adoc

```diff
@@ -485,3 +485,682 @@ Hibernate is not going to throw any exception, and it will assign a value of `nu
+
+[[associations-where-join-table-fetch-example]]
+.`@WhereJoinTable` fetch example
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/WhereJoinTableTest.java[tags=associations-where-join-table-fetch-example]
+----
+====
+
+[[associations-filter]]
+==== `@Filter`
+
+The `@Filter` annotation is another way to filter out entities or collections using custom SQL criteria.
+Unlike the `@Where` annotation, `@Filter` allows you to parameterize the filter clause at runtime.
+
+Now, considering we have the following `Account` entity:
+
+[[associations-filter-account-example]]
+.`@Filter` mapping entity-level usage
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/FilterTest.java[tags=associations-filter-Account-example]
+----
+====
+
+[NOTE]
+====
+Notice that the `active` property is mapped to the `active_status` column.
+
+This mapping was done to show you that the `@Filter` condition uses a SQL condition and not a JPQL filtering predicate.
+====
+
+As already explained, we can also apply the `@Filter` annotation for collections as illustrated by the `Client` entity:
+
+[[associations-filter-client-example]]
+.`@Filter` mapping collection-level usage
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/FilterTest.java[tags=associations-filter-Client-example]
+----
+====
+
+If we persist a `Client` with three associated `Account` entities,
+Hibernate will execute the following SQL statements:
+
+[[associations-filter-persistence-example]]
+.Persisting and fetching entities with a `@Filter` mapping
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/FilterTest.java[tags=associations-filter-persistence-example]
+----
+
+[source, SQL, indent=0]
+----
+include::{extrasdir}/associations-filter-persistence-example.sql[]
+----
+====
+
+By default, without explicitly enabling the filter, Hibernate is going to fetch all `Account` entities.
+
+[[associations-no-filter-entity-query-example]]
+.Query entities mapped without activating the `@Filter`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/FilterTest.java[tags=associations-no-filter-entity-query-example]
+----
+
+[source, SQL, indent=0]
+----
+include::{extrasdir}/associations-no-filter-entity-query-example.sql[]
+----
+====
+
+If the filter is enabled and the filter parameter value is provided,
+then Hibernate is going to apply the filtering criteria to the associated `Account` entities.
+
+[[associations-filter-entity-query-example]]
+.Query entities mapped with `@Filter`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/FilterTest.java[tags=associations-filter-entity-query-example]
+----
+
+[source, SQL, indent=0]
+----
+include::{extrasdir}/associations-filter-entity-query-example.sql[]
+----
+====
+
+[IMPORTANT]
+====
+Filters apply to entity queries, but not to direct fetching.
+
+Therefore, in the following example, the filter is not taken into consideration when fetching an entity from the Persistence Context.
```

Comment:
Should this part really be in the Associations chapter? It's not about associations as far as I can see but about entity loading.

Reply:
It is about associations just like it is about entity loading because the `@Filter` can be applied to an entity or to a collection.

Splitting it would be worse actually. Moving it entirely to a different section while keeping the `@Where` section here is also worse since these two are also related.  And `@Where` can also be applied to associations. To some extent, even `@JoinFormula` is related to the `@Where` section.

Reply:
I dunno. It's just that people won't look for entity filtering in the Associations section. That's my main concern.